cmd/csi-rclone-plugin/main.go

```diff
@@ -21,54 +23,107 @@ func init() {
-			handle()
+			handleNode()
 		},
 	}
-	cmd.AddCommand(runCmd)
-
-	runCmd.PersistentFlags().StringVar(&nodeID, "nodeid", "", "node id")
-	runCmd.MarkPersistentFlagRequired("nodeid")
-
-	runCmd.PersistentFlags().StringVar(&endpoint, "endpoint", "", "CSI endpoint")
-	runCmd.MarkPersistentFlagRequired("endpoint")
+	runNode.PersistentFlags().StringVar(&nodeID, "nodeid", "", "node id")
+	runNode.MarkPersistentFlagRequired("nodeid")
+	runNode.PersistentFlags().StringVar(&endpoint, "endpoint", "", "CSI endpoint")
+	runNode.MarkPersistentFlagRequired("endpoint")
+	runCmd.AddCommand(runNode)
+	runController := &cobra.Command{
+		Use:   "controller",
+		Short: "Start the CSI driver controller.",
+		Run: func(cmd *cobra.Command, args []string) {
+			handleController()
+		},
+	}
+	runController.PersistentFlags().StringVar(&nodeID, "nodeid", "", "node id")
+	runController.MarkPersistentFlagRequired("nodeid")
+	runController.PersistentFlags().StringVar(&endpoint, "endpoint", "", "CSI endpoint")
+	runController.MarkPersistentFlagRequired("endpoint")
+	runCmd.AddCommand(runController)
 
 	versionCmd := &cobra.Command{
 		Use:   "version",
 		Short: "Prints information about this version of csi rclone plugin",
 		Run: func(cmd *cobra.Command, args []string) {
-			fmt.Printf(`csi-rclone plugin
-Version:    %s
-`, rclone.DriverVersion)
+			fmt.Printf("csi-rclone plugin Version: %s", rclone.DriverVersion)
 		},
 	}
-	cmd.AddCommand(versionCmd)
+	root.AddCommand(versionCmd)
 
-	cmd.ParseFlags(os.Args[1:])
-	if err := cmd.Execute(); err != nil {
+	root.ParseFlags(os.Args[1:])
+	if err := root.Execute(); err != nil {
 		fmt.Fprintf(os.Stderr, "%s", err.Error())
 		os.Exit(1)
 	}
 
 	os.Exit(0)
 }
 
-func handle() {
-	kubeClient, err := kube.GetK8sClient()
+func handleNode() {
+	err := unmountOldVols()
+	if err != nil {
+		klog.Warningf("There was an error when trying to unmount old volumes: %e", err)
+	}
+	d := rclone.NewDriver(nodeID, endpoint)
+	ns, err := rclone.NewNodeServer(d.CSIDriver)
 	if err != nil {
 		panic(err)
 	}
-	d := rclone.NewDriver(nodeID, endpoint, kubeClient)
+	d.WithNodeServer(ns)
 	err = d.Run()
 	if err != nil {
 		panic(err)
 	}
 }
+
+func handleController() {
+	d := rclone.NewDriver(nodeID, endpoint)
+	cs := rclone.NewControllerServer(d.CSIDriver)
+	d.WithControllerServer(cs)
+	err := d.Run()
+	if err != nil {
+		panic(err)
+	}
+}
+
+// unmountOldVols is used to unmount volumes after a restart on a node
+func unmountOldVols() error {
+	const mountType = "fuse.rclone"
+	const unmountTimeout = time.Second * 5
+	klog.Info("Checking for existing mounts")
+	mounter := mountUtils.Mounter{}
+	mounts, err := mounter.List()
+	if err != nil {
+		return err
+	}
+	for _, mount := range mounts {
+		if mount.Type != mountType {
+			continue
+		}
+		err := mounter.UnmountWithForce(mount.Path, unmountTimeout)
+		if err != nil {
+			klog.Warningf("Failed to unmount %s because of %e, will try to unmount with force.", mount.Path, err)
```

Comment:
This message seems incorrect, as above we already unmount with force and it's not actually trying anything with this mount after this line.